modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureWriter.java

```diff
@@ -376,7 +376,13 @@ public void write() throws IOException {
         }
 
         // convert geometry
-        g = JTSUtilities.convertToCollection(g, shapeType);
+        if (g != null) {
+            if (g.isEmpty()) {
+                g = null;
+            } else {
+                g = JTSUtilities.convertToCollection(g, shapeType);
```

Comment:
It could be argued that JTSUtilities is at fault here and should return null when fed an empty geometry of any type. But I'm not sure what the knock on effects would be.

Reply:
Thought about it, but the method says "convert to collection", going from empty to null geometry seems something else to me.